imap_data_access/file_validation.py

```diff
@@ -356,3 +356,259 @@ def construct_path(self) -> Path:
+            it is assumed to be valid for all times.
+        end_time: str, optional
+            The end time for the filename. If not provided,
+            the file is valid until a file with a later
+            start_date and no end_date.
+        version : str
+            The version of the data
+
+        Returns
+        -------
+        str
+            The generated filename
+        """
+        extension = (
+            "cdf"  # TODO: double check what extensions will be, is there a list?
+        )
+        if start_time:  # TODO: are both start and end time optional in some instances?
+            time_field = start_time
+            if end_time:
+                time_field += end_time  # I think...
+            filename = (
+                f"imap_{instrument}_{ancillary_name}_{time_field}_"
+                f"{version}.{extension}"
+            )
+        else:
+            filename = f"imap_{instrument}_{ancillary_name}_" f"{version}.{extension}"
+        return cls(filename)
+
+    """Changes needed below here."""
+
+    def validate_filename(self) -> str:
+        """Validate the filename and populate the error message for wrong attributes.
+
+        The error message will be an empty string if the filename is valid. Otherwise,
+        all errors with the filename will be put into the error message.
+
+        Returns
+        -------
+        error_message: str
+            Error message for specific missing attribute, or "" if the file name is
+            valid.
+        """
+        error_message = ""
+
+        if any(
+            attr is None or attr == ""
+            for attr in [
+                self.mission,
+                self.instrument,
+                self.ancillary_name,
+                self.version,
+                self.extension,
+            ]
+        ):
+            error_message = (
+                f"Invalid filename, missing attribute. Filename "
+                f"convention is {imap_data_access.FILENAME_CONVENTION} \n"
+            )
+        if self.mission != "imap":
+            error_message += f"Invalid mission {self.mission}. Please use imap \n"
+
+        if self.instrument not in imap_data_access.VALID_INSTRUMENTS:
+            error_message += (
+                f"Invalid instrument {self.instrument}. Please choose "
+                f"from "
+                f"{imap_data_access.VALID_INSTRUMENTS} \n"
+            )
+
+        if self.extension not in imap_data_access.VALID_FILE_EXTENSION:
+            error_message += "Invalid extension. Extension should be cdf. \n"
+
+        if not self.is_valid_date(self.start_date):
+            error_message += "Invalid start date format. Please use YYYYMMDD format. \n"
+        if not self.is_valid_date(self.end_date):
+            error_message += "Invalid end date format. Please use YYYYMMDD format. \n"
+
+        return error_message
+
+    @staticmethod
+    def is_valid_date(input_date: str) -> bool:
```

Comment:
Should use the version from ScienceFilePath rather than duplicating it. Since that method is static, you can replace any calls to `self.is_valid_date()` with `ScienceFilePath.is_valid_date()` and it will use the already-existing method (that is why it is static in ScienceFilePath.)

If you want to use a non-static method, you can inherit the class and methods and then use it that way. Have you taken a class in object oriented coding? I am happy to hop on a call and explain some of the basics.